### PR TITLE
APS Logout

### PR DESCRIPTION
**Pull request by azaleacolburn**
### Description
Allows logout from your APS account in fission.

### TODO:
- [x] Write functions to logout

### Testing Done
- [x] Test logout

[JIRA Issue](https://jira.autodesk.com/browse/AARD-1709)

### CODE REVIEW

**Comment by HunterBarclay** [comment]
So, quick note. This ticket was actually meant for Fission, not Fusion. Regardless, both should be done so could you extend this PR into also adding proper revoking capabilities to Fission as well?

**Review by BrandonPacewic — COMMENTED** [review]
@Ewie21 Ready for review? Looks like there are still some `TODO`s unchecked.

File: fission/src/aps/APS.ts [hunk 1]
```diff
@@ -122,12 +123,43 @@ class APS {
     }
 
     /**
-     * Logs the user out by setting their auth data to undefined.
+     * Logs the user out by setting their auth data to undefined and revoking their auth token.
+     * Does nothing if the auth token couldn't be revoked
      */
     static async logout() {
+        if (!(await this.revoke_token_public())) return
```
Comment (BrandonPacewic) [inline fission/src/aps/APS.ts]:
```suggestion
        if (!(await this.revokeTokenPublic())) return
```

File: fission/src/aps/APS.ts [hunk 2]
```diff
@@ -122,12 +123,43 @@ class APS {
...
     }
 
+    /*
+     * Revokes the users token
+     *
+     * The client should be public since we're an spa
+     * Endpoint documentation:
+     * https://aps.autodesk.com/en/docs/oauth/v2/reference/http/revoke-POST/
+     */
+    static async revoke_token_public(): Promise<boolean> {
```
Comment (BrandonPacewic) [inline fission/src/aps/APS.ts]:
```suggestion
    static async revokeTokenPublic(): Promise<boolean> {
```

**Review by BrandonPacewic — CHANGES_REQUESTED** [review]
Looks good! One minor request is that we stick to camel case for function names.

File: fission/bun.lockb
Comment (HunterBarclay) [inline fission/bun.lockb:1]: Can we remove this. I believe it is in the git ignore in dev.

File: fission/src/aps/APS.ts (hunk 1 above)
Comment (HunterBarclay) [inline fission/src/aps/APS.ts]: If the token isn't revoked, lets just drop it anyways.

**Review by HunterBarclay — CHANGES_REQUESTED** [review]
Revoked great, couple changes.

File: fission/src/aps/APS.ts [hunk 3]
```diff
@@ -122,12 +123,43 @@ class APS {
     }
 
     /**
-     * Logs the user out by setting their auth data to undefined.
+     * Logs the user out by setting their auth data to undefined and revoking their auth token.
+     * Does nothing if the auth token couldn't be revoked
```
Comment (BrandonPacewic) [inline fission/src/aps/APS.ts]: Should this comment line be removed now that we always set the auth token to undefined even if we can't revoke the token?

```suggestion
```

**Review by BrandonPacewic — APPROVED** [review]
Revoking my stale review...

**Review by PepperLola — APPROVED** [review]
Looks like it works well. For some reason I'm still able to get my own user info after revoking the token but nothing else works (which is good).

**Review by HunterBarclay — APPROVED** [review]

**Pull request merged**